x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticationToken.java

```diff
@@ -35,9 +35,8 @@
      * @param clientAuthenticationSharedSecret URL-safe Shared Secret for Client authentication. Required by some JWT realms.
      */
     public JwtAuthenticationToken(final SecureString endUserSignedJwt, @Nullable final SecureString clientAuthenticationSharedSecret) {
-        if (endUserSignedJwt == null) {
-            throw new IllegalArgumentException("JWT bearer token must be non-null");
-        } else if (endUserSignedJwt.isEmpty()) {
+        assert endUserSignedJwt != null;
```

Comment:
Does it make sense to revert this change? I am not sure if it is needed.

The null check in token() seems to be sufficient to fix the continuation bug going from JWT realm to non-JWT realm.
https://github.com/elastic/elasticsearch/blob/799a681906bfa35b8b94cd41b6a1e12625ff05d0/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java#L291-L293

By changing `JwtAuthenticationToken` constructor, null JWT will still fail, but the descriptive error message is gone.
- Test: Null JWT triggers assert failure without the original descriptive error message.
- Main: Null JWT falls through and triggers an uncaught NullPointerException without the original descriptive error message on this line.
https://github.com/elastic/elasticsearch/blob/799a681906bfa35b8b94cd41b6a1e12625ff05d0/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticationToken.java#L49

Reply:
I can remove the assert of course. It's just a stylistic choice than anything else, I tend to use them.